datastore/db/bytebase/FertiScan/new_inspection_function.sql

```diff
@@ -233,28 +236,36 @@ BEGIN
 
 -- SUB LABELS
 	-- Loop through each sub_type
-    FOR sub_type_rec IN SELECT id,type_en FROM sub_type
-    LOOP
+   FOR sub_type_rec IN SELECT id, type_en FROM sub_type
+	LOOP
 		-- Extract the French and English arrays for the current sub_type
-
 		key_string := sub_type_rec.type_en;
-        en_values := input_json -> key_string -> 'en';
-    	fr_values := input_json -> key_string -> 'fr';
-		key_string := key_string || '_ids';
-        -- Ensure both arrays are of the same length
-        IF jsonb_array_length(fr_values) = jsonb_array_length(en_values) THEN
-		  	FOR i IN 0..(jsonb_array_length(fr_values) - 1)
-	   		LOOP
-	   			sub_label_id := "fertiscan_0.0.12".new_sub_label(
-					fr_values->>i,
-					en_values->>i,
-					label_info_id,
-					sub_type_rec.id,
-					FALSE
-				);
-			END LOOP;
-		END IF;
-   END LOOP;    
+		en_values := COALESCE(input_json -> key_string -> 'en', '[]'::jsonb);
+		fr_values := COALESCE(input_json -> key_string -> 'fr', '[]'::jsonb);
+
+		-- Determine the maximum length of the arrays
+		max_length := GREATEST(
```

Comment:
Do you think we should raise a notice if the length is not equal? This could be some information left in the comment section of the JSON between layers.

(this could be a future issue)

Reply:
Done.